arcaflow_plugin_fio/fio_schema.py

```diff
@@ -174,128 +228,153 @@ class JobParams:
         schema.description(
             "Don't pre-create files but allow the job's open() to create a file when "
-            "it's time to do I/O."
+            "it's time to do I/O. Defaults to false."
         ),
     ] = None
     pre_read: typing.Annotated[
-        typing.Optional[int],
-        validation.min(0),
-        validation.max(1),
+        typing.Optional[bool],
         schema.name("Pre-Read Files"),
         schema.description(
             "Files will be pre-read into memory before starting the given I/O "
             "operation. This will also clear the invalidate flag, since it is "
             "pointless to pre-read and then drop the cache. This will only work for "
             "I/O engines that are seek-able, since they allow you to read the same "
             "data multiple times. Thus it will not work on non-seekable I/O engines "
-            "(e.g. network, splice)."
+            "(e.g. network, splice). Defaults to false."
         ),
     ] = None
     unlink: typing.Annotated[
-        typing.Optional[int],
-        validation.min(0),
-        validation.max(1),
+        typing.Optional[bool],
         schema.name("Unlink Files"),
         schema.description(
             "Unlink the job files when done. Not the default, as repeated runs of that "
-            "job would then waste time recreating the file set again and again."
+            "job would then waste time recreating the file set again and again. "
+            "Defaults to false."
         ),
     ] = None
     unlink_each_loop: typing.Annotated[
-        typing.Optional[int],
-        validation.min(0),
-        validation.max(1),
+        typing.Optional[bool],
         schema.name("Unlink Files Each Loop"),
-        schema.description("Unlink job files after each iteration or loop."),
+        schema.description(
+            "Unlink job files after each iteration or loop. Defaults to False."
+        ),
     ] = None
 
     # I/O Type
     direct: typing.Annotated[
-        typing.Optional[int],
-        validation.min(0),
-        validation.max(1),
+        typing.Optional[bool],
         schema.name("Direct I/O"),
         schema.description(
             "Use non-buffered I/O. This is usually O_DIRECT. Note that OpenBSD and ZFS "
             "on Solaris don't support direct I/O. On Windows the synchronous ioengines "
-            "don't support direct I/O."
+            "don't support direct I/O. This is the opposite of the buffered option and "
+            "defaults to False."
         ),
     ] = None
     buffered: typing.Annotated[
-        Optional[int],
-        validation.min(0),
-        validation.max(1),
+        Optional[bool],
         schema.name("Buffered"),
         schema.description(
-            "Use buffered I/O. This is the opposite of the direct option."
+            "Use buffered I/O. This is the opposite of the direct option and defaults "
+            "to True."
         ),
     ] = None
```

Comment:
Should we be using `true` instead of `None` at line 285?

Reply:
I don't think so. All of the defaults come from the underlying fio tool, not from the plugin schema. Setting this to `true` pushes that value explicitly into the fio input file unnecessarily.